### ISSUE TITLE
when training 32%, the multiprocess counted error

### ISSUE THREAD

**Issue by VanHulk** [Liuziyu77/Visual-RFT#72]
I don't understand why the training was interrupted during the training process. It seems work well before. 
We use A100(40g) *  4.

The scripts  is:  
export DATA_PATH= Visual-RFT/huggingface/ViRFT_COCO_base65
export CKPT_PATH=/data/LLM_model/Qwen2-VL-2B-Instruct/Qwen2-VL-2B-Instruct
export SAVE_PATH=/data/LLM_model/Qwen2-VL-2B-Instruct/Qwen2-VL-2B-Instruct_GRPO_coco_base65

export DEBUG_MODE="true" # Enable Debug if you want to see the rollout of model during RL
export LOG_PATH="./debug_log_2b_GRPO_coco_base65.txt"

export WANDB_NAME="Visaul-RFT-GRPO-2B-COCO-base65"
export WANDB_NOTES="Smaller learning rate, more regularization."
export PYTORCH_CUDA_ALLOC_CONF="expandable_segments:True,max_split_size_mb:128"

torchrun --nproc_per_node="4" \
    --nnodes="1" \
    --node_rank="0" \
    --master_addr="127.0.0.1" \
    --master_port="12345" \
    ./src/open_r1/grpo.py \
    --output_dir ${SAVE_PATH}  \
    --model_name_or_path ${CKPT_PATH} \
    --dataset_name ${DATA_PATH} \
    --deepspeed ./local_scripts/zero3.json \
    --max_prompt_length 1024 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 2 \
    --logging_steps 1 \
    --bf16 \
    --report_to wandb \
    --gradient_checkpointing False \
    --attn_implementation flash_attention_2 \
    --max_pixels 401408 \
    --num_train_epochs 2 \
    --run_name Qwen2-VL-2B_GRPO_coco_base65\
    --save_steps 100 \
    --save_only_model true \
    --num_generations 4

![Image](https://github.com/user-attachments/assets/4aba71b3-cf2e-43af-a3dc-c4f3992d5245)

**Comment by Ustinianer** [comment]
I had a similar problem. Did you solve it？

**Comment by Ustinianer** [comment]
I tried to add a reward for the length of think and it seems to work, u can check it out in this link：
https://github.com/Liuziyu77/Visual-RFT/issues/22#issuecomment-2728586362

**Comment by 2019211753** [comment]
same

**Comment by zhengli97** [comment]
I also encountered the same problem.

**Comment by buptyqx** [comment]
I also encountered the same problem.

**Comment by BailiangJ** [comment]
check [handle NaN confidence](https://github.com/Liuziyu77/Visual-RFT/issues/100#issuecomment-2746998369) in issue #100
